### PR TITLE
Pr attention target nav

### PR DESCRIPTION
**Pull request by le-horizon**
Hey Wei and Haonan,

Adding attention to target navigation plus other supporting changes.  Attention isn't successful yet, but the change is getting large, so pushing it to avoid all the rebasing.

Could you take a look?

Thanks,
Le

### CODE REVIEW

File: alf/utils/common.py [hunk 1]
```diff
@@ -205,7 +205,7 @@ def _get_summary_enabled_var():
     global _summary_enabled_var
     if _summary_enabled_var is None:
         _summary_enabled_var = tf.Variable(
-            False, dtype=tf.bool, name="summary_enabled")
+            False, dtype=tf.bool, trainable=False, name="summary_enabled")
```
Comment (le-horizon) [inline alf/utils/common.py]: Making it non-trainable, as otherwise model checkpointing would fail.

File: alf/utils/summary_utils.py [hunk 2]
```diff
@@ -136,7 +136,8 @@ def histogram_continuous(name,
             bucket_counts = tf.cast(
                 tf.reduce_sum(input_tensor=one_hots, axis=0), dtype=tf.float64)
             edges = tf.linspace(bucket_min, bucket_max, bucket_count + 1)
-            edges = tf.concat([edges[:-1], [float(bucket_max)]], 0)
+            edges = tf.concat([edges[:-1], [tf.cast(bucket_max, tf.float64)]],
```
Comment (le-horizon) [inline alf/utils/summary_utils.py]: summarizing in a tf.function, float doesn't work, has to use tf.cast explicitly.

File: alf/environments/suite_socialbot.py [hunk 3]
```diff
@@ -74,6 +74,10 @@ def load(environment_name,
         A PyEnvironmentBase instance.
     """
     _unwrapped_env_checker_.check_and_update(wrap_with_process)
+    if gym_env_wrappers is None:
+        gym_env_wrappers = ()
+    if env_wrappers is None:
+        env_wrappers = ()
```
Comment (emailweixu) [inline alf/environments/suite_socialbot.py]: Why do you need this? The default is already `()`

File: alf/utils/common.py [hunk 4]
```diff
@@ -708,8 +708,8 @@ def get_states_shape():
         Returns 0 if internal states is not part of observation.
     """
     assert _env, "set a global env by `set_global_env` before using the function"
-    assert isinstance(_env.observation_spec(), dict), "observation not a dict"
-    if 'states' in _env.observation_spec():
+    if isinstance(_env.observation_spec(),
```
Comment (emailweixu) [inline alf/utils/common.py]: should raise if states is not part of the observation?

File: alf/utils/common.py [hunk 5]
```diff
@@ -737,8 +742,11 @@ def get_vocab_size():
         Returns 0 if language is not part of observation.
     """
     assert _env, "set a global env by `set_global_env` before using the function"
-    if 'sentence' in _env.observation_spec():
-        return _env.observation_spec()['sentence'].shape[0]
+    if isinstance(_env.observation_spec(),
```
Comment (emailweixu) [inline alf/utils/common.py]: should raise if there is not vocab_size

File: alf/environments/suite_socialbot.py (hunk 3 above)
Comment (le-horizon) [reply alf/environments/suite_socialbot.py]: This is for command line argument override for gin configs.
--gin_param="xxx=()"
Parens can be mis-interpreted by the linux shell, passing in None is easier.

File: alf/utils/common.py (hunk 4 above)
Comment (le-horizon) [reply alf/utils/common.py]: Good point.  Done.

File: alf/utils/common.py (hunk 5 above)
Comment (le-horizon) [reply alf/utils/common.py]: Done.

**Pull request merged**